phpdotnet/phd/Config.php

```diff
@@ -5,49 +5,51 @@ class Config
 {
     const VERSION = '@phd_version@';
 
-    private static $optionArrayDefault = array(
-        'output_format'     => array(),
-        'no_index'          => false,
-        'force_index'       => false,
-        'no_toc'            => false,
-        'xml_root'          => '.',
-        'xml_file'          => './.manual.xml',
-        'history_file'      => './fileModHistory.php',
-        'lang_dir'          => './',
-        'language'          => 'en',
-        'fallback_language' => 'en',
-        'verbose'           => VERBOSE_DEFAULT,
-        'date_format'       => 'H:i:s',
-        'render_ids'        => array(
-        ),
-        'skip_ids'          => array(
-        ),
-        'color_output'      => true,
-        'output_dir'        => './output/',
-        'output_filename'   => '',
-        'php_error_output'  => STDERR,
-        'php_error_color'   => '01;31', // Red
-        'user_error_output' => STDERR,
-        'user_error_color'  => '01;33', // Yellow
-        'phd_info_output'   => STDOUT,
-        'phd_info_color'    => '01;32', // Green
-        'phd_warning_output' => STDOUT,
-        'phd_warning_color' => '01;35', // Magenta
-        'highlighter'       => 'phpdotnet\\phd\\Highlighter',
-        'package'           => array(
-            'Generic',
-        ),
-        'css'               => array(),
-        'process_xincludes' => false,
-        'ext'               => null,
-        'package_dirs'      => array(__INSTALLDIR__),
-        'saveconfig'        => false,
-        'quit'              => false,
-        'indexcache'        => '',
-        'memoryindex'       => '',
-    );
-
-    private static $optionArray = null;
+    /** @var array<string, string> */
+    public array $output_format = [];
```

Comment:
Is the change from static to instance properties okay? (not that I think having it static in the first place made a lot of sense).

Reply:
I think it makes sense that one set of configuration options can be assigned to one object of the class Config and then we just pass the appropriate object when needed.

As far as how Config is used in PhD, this doesn't make much of a difference as only one Config object is used throughout. :-)